### PR TITLE
ANDROAPP-5495-mobile-ui-Button-Carrousel

### PR DESCRIPTION
**Pull request by DavidAparicioAlbaAsenjo**

### CODE REVIEW

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ButtonCarousel.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+@Composable
+fun ButtonCarousel(
+    carouselButtonList: List<CarouselButtonData>,
+    onClick: () -> Unit
+) {
+   FlowRowComponentsContainer(
+       content = {
+           carouselButtonList.map {
+               CarouselButton(
+                   enabled = it.enabled,
```
Comment (xavimolloy) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ButtonCarousel.kt:77]: How do you differentiate between one onclick event and another?  right now they all share the same onClick event

**Review by xavimolloy — CHANGES_REQUESTED** [review]

File: common/src/commonMain/kotlin/org/hisp/dhis/common/App.kt [hunk 2]
```diff
@@ -24,10 +24,11 @@ import androidx.compose.ui.graphics.Color
 import org.hisp.dhis.common.screens.BadgesScreen
 import org.hisp.dhis.common.screens.BottomSheetHeaderScreen
 import org.hisp.dhis.common.screens.ButtonBlockScreen
+import org.hisp.dhis.common.screens.ButtonCarouselScreen
 import org.hisp.dhis.common.screens.ButtonScreen
 import org.hisp.dhis.common.screens.CheckboxScreen
 import org.hisp.dhis.common.screens.ChipsScreen
-import org.hisp.dhis.common.screens.Components
+import org.hisp.dhis.common.screens.Components.*
```
Comment (andresmr) [inline common/src/commonMain/kotlin/org/hisp/dhis/common/App.kt:31]: avoid using wildcards, I think this will fail the checkstyle

**Review by andresmr — CHANGES_REQUESTED** [review]

File: common/src/commonMain/kotlin/org/hisp/dhis/common/App.kt (hunk 2 above)
Comment (DavidAparicioAlbaAsenjo) [reply common/src/commonMain/kotlin/org/hisp/dhis/common/App.kt]: done!

**Review by andresmr — APPROVED** [review]

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ButtonCarousel.kt (hunk 1 above)
Comment (DavidAparicioAlbaAsenjo) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/ButtonCarousel.kt]: now it's passed in the component directly

**Review by xavimolloy — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Pull request merged**